### PR TITLE
Add the missing cursor parameter to unified search requests

### PR DESCRIPTION
**Pull request by ChristophWurst**
Otherwise paginated search does not work. Basically you always requested
the first x items and then showed them again and again. The second or
any other subsequent page was never requested.

On a side note this also caused lots of Vue warning due to duplicate
keys.

### CODE REVIEW

File: core/src/services/UnifiedSearchService.js [hunk 1]
```diff
@@ -53,8 +53,13 @@ export async function getTypes() {
...
  * @param {string} query the search
+ * @param {int|string|undefined} cursor the offset for paginated searches
  * @returns {Promise}
  */
-export function search(type, query) {
-	return axios.get(generateUrl(`/search/providers/${type}/search?term=${query}`))
+export function search(type, query, cursor) {
+	return axios.get(generateUrl(`/search/providers/${type}/search?term=${query}`), {
+		params: {
+			cursor,
```
Comment (nickvergessen) [inline core/src/services/UnifiedSearchService.js:62]:
```suggestion
			cursor,
			from: window.location.pathname.replace('/index.php', '') + window.location.search,
```

this is missing the from too?

Comment (nickvergessen) [reply core/src/services/UnifiedSearchService.js:62]: Ref https://github.com/nextcloud/server/blob/master/core/Controller/UnifiedSearchController.php#L98

Comment (ChristophWurst) [reply core/src/services/UnifiedSearchService.js:62]: possibly. feel free adding it here :)

**Review by skjnldsv — APPROVED** [review]
Wut, this was working before! :thinking:

**Review by juliusknorr — APPROVED** [review]

**Comment by nickvergessen** [comment]
> Conflicting files
core/js/dist/unified-search.js
core/js/dist/unified-search.js.map

:roll_eyes:

**Pull request merged**